### ISSUE TITLE
Define a clear privacy policy

### ISSUE THREAD

**Issue by insertapp** [PretendoNetwork/website#296]
_Apologies if this is the wrong repository_
As I was planning on registering a pretendo account I noticed there was no defined privacy policy or terms of service to inform me of how my data will be used if I register. Looking at other projects (Wiimifi) a clear privacy policy is defined in order to inform users of how their data is used and their right to erasure. As pretendo becomes more popular I believe this is something that must be addressed.

**Comment by ExperiencersInternational** [comment]
Yeah, I agree in regards with compliance with GDPR/CCPA.

Can't find any examples here either, I'm sure there's a Terms of Use somewhere but I can't find that easily either, need to be prominent links in the footer if they're not already.

**Comment by jonbarrow** [comment]
> *Apologies if this is the wrong repository*

This is 100% the right place, no worries :+1:

> As I was planning on registering a pretendo account I noticed there was no defined privacy policy or terms of service to inform me of how my data will be used if I register
> ...
> I believe this is something that must be addressed

We are aware of this and we (hopefully obviously!) agree! It's being fixed ASAP. As an American without a lawyer and limited knowledge in European law, I wanted to be sure the policy was both robust and clear enough, while remaining compliant. Originally I was talking with @InternalLoss about this (as he's from across the pond) to ensure things were good before plastering it everywhere, but stuff just came up. However it's being reviewed again as we speak and should be prominent soon.

That being said, you can see exactly how your data is used by looking at our source code. The code we publish here is exactly what is deployed, so you can see how it's stored/managed, and we don't do anything with that data outside of what is strictly necessary for operating the services (while also collecting the bare minimum). We don't even log IPs or anything on our servers.

**Comment by ExperiencersInternational** [comment]
> We don't even log IPs or anything on our servers.

Just want to take out this bit specifically Jon, the Discourse server logs IPs right?

I recall @.ashquarky saying that on the forum last night as a reason why they can't have TL4 (I think they were confusing it with administrator permissions). I do recall Discourse logging IPs when I was an admin for a forum, iirc it's to assist with IP bans?

Might just be a misunderstanding from both our ends.

**Comment by jonbarrow** [comment]
> Just want to take out this bit specifically Jon, the Discourse server logs IPs right?

Correct, I wasn't clear. I meant that anything we produce does not track IPs. We don't produce Discourse so I didn't think about it. That being said, the Discourse policy would apply here I believe? https://www.discourse.org/privacy

I've looked for a way to disable this but Discourse does not seem to provide that option, nor do they seem to consider IP addresses PII? I was in conversations with Discourse staff recently regarding our integration requirements, and I was told by Discourse that admins are the only ones able to see peoples PII. Given that users with TL4, regardless of admin rights, can see IP addresses, it seems that they do not consider them to be PII? Relevant screenshot and link:

https://meta.discourse.org/t/integration-into-custom-auth-system-where-emails-are-not-unique/306489/22

![Screenshot from 2024-05-11 14-05-31](https://github.com/PretendoNetwork/website/assets/27011796/67a09d74-63db-47ed-a825-2c85667fe7b2)

**Comment by ExperiencersInternational** [comment]
> > Just want to take out this bit specifically Jon, the Discourse server logs IPs right?
> 
> Correct, I wasn't clear. I meant that anything we produce does not track IPs. We don't produce Discourse so I didn't think about it. That being said, the Discourse policy would apply here I believe? https://www.discourse.org/privacy
> 
> I've looked for a way to disable this but Discourse does not seem to provide that option, nor do they seem to consider IP addresses PII? I was in conversations with Discourse staff recently regarding our integration requirements, and I was told by Discourse that admins are the only ones able to see peoples PII. Given that users with TL4, regardless of admin rights, can see IP addresses, it seems that they do not consider them to be PII? Relevant screenshot and link:
> 
> https://meta.discourse.org/t/integration-into-custom-auth-system-where-emails-are-not-unique/306489/22
> 
> ![Screenshot from 2024-05-11 14-05-31](https://private-user-images.githubusercontent.com/27011796/329805342-67a09d74-63db-47ed-a825-2c85667fe7b2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU0NzM0MzYsIm5iZiI6MTcxNTQ3MzEzNiwicGF0aCI6Ii8yNzAxMTc5Ni8zMjk4MDUzNDItNjdhMDlkNzQtNjNkYi00N2VkLWE4MjUtMmM4NTY2N2ZlN2IyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTEyVDAwMTg1NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkwYzg4Y2Q2NDZhMjhmYTFkNzExNDVkZGQyZjAzNGIwOTY3MjlkMzczZDkwMTkyMGY3YTQwYmM1NGVlY2I0NGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.oVRp1VkYlK-3j9IemwCVsdMES3aVBh6U3Mwf2yDb9GM)

If I recall correctly, Discourse does automatically create a privacy policy/ToU for your project on installation and I think you're supposed to refer to that?

I don't honestly see the IP addresses being too sensitive, especially when most people will be accessing over a dynamic IP.

**Comment by ExperiencersInternational** [comment]
Any updates on this? Seeing a lot of people requesting account deletions on the forum but there is no easy way to do that yet afaik.

**Comment by Rathmox** [comment]
> Any updates on this? Seeing a lot of people requesting account deletions on the forum but there is no easy way to do that yet afaik.

@ExperiencersInternational User who want to delete his account here.

There is still no easy way, the website cannot do it and the deletion process from the console cannot either.

Also what I ask a lot for is changing the PNID, I don't think it's easy but it would be better if it was possible (Nintendo not allowing this is against GDPR article 16 afaik)

**Comment by jonbarrow** [comment]
> Any updates on this? Seeing a lot of people requesting account deletions on the forum but there is no easy way to do that yet afaik.

Yes. We are still updating the documents, as to be entirely thorough with what we do and do not use and how it is used, and to make sure we are compliant. These sorts of changes also require changes to several services at once, not just here, and @hauntii is looking into the website rewrite which would include these changes. Below is an example of one of his draft implementations from this rewrite:

![image (25)](https://github.com/user-attachments/assets/c5cceeea-84be-4b9f-a5f6-cf60a0369e10)

That being said, accounts *can* be deleted either from a console (the endpoints *are* implemented) or by contacting us directly. So while not as easy as just clicking a button, it is far from impossible

**Comment by jonbarrow** [comment]
> the deletion process from the console cannot either.

This is incorrect. The account deletion endpoints do exist on the console API server and do function. If you are having issues with this, then you should open an issue report on the appropriate repository

> Also what I ask a lot for is changing the PNID, I don't think it's easy but it would be better if it was possible (Nintendo not allowing this is against GDPR article 16 afaik)

This is functionally impossible due to the way Nintendo Network is designed. There are parts of games/servers which require the use of unique, never changing, usernames. This is something which cannot be worked around. That being said, there are plenty of other services which also do not allow you to change usernames once set, including

- Reddit (https://support.reddithelp.com/hc/en-us/articles/204579479-Can-I-change-my-username)
- League of Legends (https://support-leagueoflegends.riotgames.com/hc/en-us/articles/201752814-Riot-Account-Names-FAQ)
- Steam (https://help.steampowered.com/en/faqs/view/2816-BE67-5B69-0FEC)

They typically allow you to change something like a nickname, but not your account ID/username. This is also the case with us, where your Mii/Mii name is what is commonly used in games

**Comment by Rathmox** [comment]
> This is functionally impossible due to the way Nintendo Network is designed.
I kinda expected it

> This is incorrect. The account deletion endpoints do exist on the console API server and do function. If you are having issues with this, then you should open an issue report on the appropriate repository
3ds, I get error code 005-4270
https://github.com/PretendoNetwork/account/issues/102

**Comment by jonbarrow** [comment]
We apologize for the long wait. We had to wait for things such our LLC to be created/renewed (as it is mentioned in the policies), and I've been making tweaks to it pretty consistently as I wasn't always happy with how the documents turned out. Eventually we decided to just use a service to create the base of the documents for us, and then modify them to our needs, which you can now find here:

https://github.com/PretendoNetwork/website/pull/377

We are not legal experts, so we hope the terms are satisfactory, but if not please do feel free to comment

**Comment by jonbarrow** [comment]
The policies are now live on our website. I went ahead and merged the PR since there were no objections to the wording, and I wanted to get this pushed out the door asap. If there are any future issues or questions, we can definitely address them as they come up. For now I think what we have is good though (the only "complaint" we had was the amount of legalese, but that's the cost we pay for using a service to make these documents for us)

https://pretendo.network/terms/privacy
